database.ts

```diff
@@ -258,41 +245,26 @@ export async function originalEventToParsedEvent(
     if (publicKey instanceof Error) {
         return publicKey;
     }
-
     const parsedTags = getTags(event);
-    let e: Text_Note_Event | Encrypted_Event | Profile_Nostr_Event;
     if (event.kind == NostrKind.CustomAppData || event.kind == NostrKind.DIRECT_MESSAGE) {
-        const _e = await originalEventToEncryptedEvent(
-            {
-                ...event,
-                kind: event.kind,
-            },
+        return originalEventToEncryptedEvent(
+            event,
             ctx,
             parsedTags,
             publicKey,
             eventsRemover,
         );
-        if (_e instanceof Error || _e == false) {
-            return _e;
-        }
-        e = _e;
+        // return false
     } else if (event.kind == NostrKind.META_DATA || event.kind == NostrKind.TEXT_NOTE) {
-        const _e = originalEventToUnencryptedEvent(
-            {
-                ...event,
-                kind: event.kind,
-            },
+        return originalEventToUnencryptedEvent(
+            // @ts-ignore
+            event,
```

Comment:
Instead of pleasing the compiler, we can skip the type check because we know it's correct.